src/parse.rs

```diff
@@ -105,21 +88,37 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_get_file_extension_from_diff_line() {
+    fn test_get_file_extension_from_marker_line() {
         assert_eq!(
-            get_file_extension_from_diff_line("diff --git a/src/main.rs b/src/main.rs"),
+            get_file_extension_from_marker_line(
+                "--- src/one.rs	2019-11-20 06:47:56.000000000 +0100"
+            ),
             Some("rs")
         );
     }
 
     #[test]
-    fn test_get_file_extension_from_marker_line() {
+    fn test_get_file_extension_from_file_meta_line() {
         assert_eq!(
-            get_file_extension_from_marker_line(
-                "--- src/one.rs	2019-11-20 06:47:56.000000000 +0100"
-            ),
+            get_file_extension_from_file_meta_line_file_path("a/src/parse.rs"),
             Some("rs")
         );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("b/src/pa rse.rs"),
+            Some("rs")
+        );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("src/pa rse.rs"),
+            Some("rs")
+        );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("wat hello.rs"),
+            Some("rs")
+        );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("/dev/null"),
+            None
+        );
```

Comment:
Would you mind adding test cases for something like `Makefile` that lacks an extension? (Would it be cleaner for it to be handled by the same code path as `/dev/null`?)

Reply:
Added the test cases. 
`/dev/null` seems to be a special case to me so I made a separate code path for it but if you prefer to handle a file without extension in the same code path, I can make the change.

Reply:
Thanks!